srtp/srtp.c

```diff
@@ -193,56 +214,74 @@ srtp_stream_alloc(srtp_stream_ctx_t **str_ptr,
   memset(str, 0, sizeof(srtp_stream_ctx_t));
   *str_ptr = str;
 
-  /* allocate cipher */
-  stat = srtp_crypto_kernel_alloc_cipher(p->rtp.cipher_type, 
-				    &str->rtp_cipher, 
-				    p->rtp.cipher_key_len,
-				    p->rtp.auth_tag_len); 
-  if (stat) {
-    srtp_stream_free(str);
-    return stat;
+  /* To keep backwards API compatible if someone is using multiple master
+   * keys then key should be set to NULL
+   */
+  if (p->key != NULL) {
+      str->num_master_keys = 1;
+  } else {
+      str->num_master_keys = p->num_master_keys;
   }
 
-  /* allocate auth function */
-  stat = srtp_crypto_kernel_alloc_auth(p->rtp.auth_type, 
-				  &str->rtp_auth,
-				  p->rtp.auth_key_len, 
-				  p->rtp.auth_tag_len); 
-  if (stat) {
-    srtp_stream_free(str);
-    return stat;
+  str->session_keys = (srtp_session_keys_t *)srtp_crypto_alloc(sizeof(srtp_session_keys_t) * str->num_master_keys);
```

Comment:
This function does not set memory to zero.  I would suggest doing a memset() here so that, in case there is a failure, the data structure will be zerod and the srtp_stream_free() will not misinterpret random values are valid values.